app/hooks/stateModifiers.tsx

```diff
@@ -5,11 +5,25 @@ type StateString = string
 
 export const TOP = '-1'
 
+const uniqueChecker = (): ((e: StateEntry) => boolean) => {
+  const seen = new Set()
+  return ([pageid]) => !seen.has(pageid) && Boolean(seen.add(pageid))
+}
 export const getStateEntries = (state: StateString): StateEntry[] =>
-  Array.from(state.matchAll(/([^-_r]+)([-_r]*)/g) ?? []).map((groups) => [
-    groups[1], // question id
-    (groups[2] as QuestionState) || QuestionState.OPEN,
-  ])
+  Array.from(state.matchAll(/([^-_r]+)([-_r]*)/g) ?? [])
+    .map(
+      (groups) =>
+        [
+          groups[1], // question id
+          (groups[2] as QuestionState) || QuestionState.OPEN,
+        ] as StateEntry
+    )
+    .filter(uniqueChecker())
```

Comment:
make sure that duplicates get removed from the url string. The first one found is the value (and position) in the resulting url string